### PR TITLE
refactor: fix improve favourites bugs

### PR DESCRIPTION
**Pull request by Nisha1306**
Closes https://github.com/AtB-AS/kundevendt/issues/2592
https://github.com/AtB-AS/kundevendt/issues/2592#issuecomment-1303217473

### CODE REVIEW

File: src/components/touchable-opacity-or-view/TouchableOpacityOrView.tsx [hunk 1]
```diff
@@ -5,13 +5,22 @@ const TouchableOpacityOrView = ({
...
   onClick,
   children,
+  accessibilityHint,
+  accessibilityLabel,
 }: {
   style: ViewStyle;
   children: ReactNode;
   onClick?: () => void;
+  accessibilityHint?: string;
+  accessibilityLabel?: string;
```
Comment (gorandalum) [inline src/components/touchable-opacity-or-view/TouchableOpacityOrView.tsx:15]: A `View` can have also have accessibility fields. Might be better to just add the `AccessibilityProps`-type here, and if provided add to both `View` and `TouchableOpacity`?

File: src/screens/Departures/PlaceScreen.tsx [hunk 2]
```diff
@@ -122,7 +130,7 @@ export default function PlaceScreen({
           <Button
             interactiveColor="interactive_0"
             text={t(DeparturesTexts.closeButton.label)}
-            onPress={() => navigation.popToTop()}
+            onPress={() => navigation.pop(selectedQuay ? 3 : 2)} // check if there is a smart way of doing this
```
Comment (gorandalum) [inline src/screens/Departures/PlaceScreen.tsx:133]: We need to look at this 😅

File: src/screens/Departures/components/EstimatedCallItem.tsx [hunk 3]
```diff
@@ -179,7 +195,18 @@ function getA11yLineLabel(departure: any, t: any, language: any) {
 
   return `${
     departure.cancellation ? t(CancelledDepartureTexts.message) : ''
-  } ${a11yLine} ${a11yFrontText} ${a11yDateInfo}`;
+  } ${getLineA11yLabel(departure, t)} ${a11yDateInfo}`;
+}
+
+function getLineA11yLabel(departure: any, t: any) {
```
Comment (gorandalum) [inline src/screens/Departures/components/EstimatedCallItem.tsx:201]: I see `any` was used here from before, but why? 🤔 I think we can try to remove them now that we are touching the code.

Comment (Nisha1306) [reply src/screens/Departures/components/EstimatedCallItem.tsx:201]: yes :)

File: src/screens/Departures/PlaceScreen.tsx (hunk 2 above)
Comment (gorandalum) [reply src/screens/Departures/PlaceScreen.tsx:133]: Tried `popToTop`?

**Review by gorandalum — APPROVED** [review]

**Pull request merged**